### PR TITLE
Solar array deployment scenario

### PR DESCRIPTION
**Pull request by leahkiner**
* **Tickets addressed:** bsk-646
* **Review:** By commit 
* **Merge strategy:** Merge (no squash) 

## Description
This PR adds a solar array deployment example script to Basilisk.

## Verification
No tests are used to validate this example script, however plots are added to confirm the solar arrays are deploying as expected.

## Documentation
No documentation is added to this PR

## Future work
In the future the array element inertias should be changed to more accurately represent triangular wedges.

### CODE REVIEW

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down: You can delete this comment about author and date. The git history captures all this information.

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down:  Let's reduce these blocks by using `map` and a lambda. E.g. 

```
array1DataLogs = list(map(lambda profiler: profiler.spinningBodyOutMsg.recorder(dataRecRate),
                              array1RotProfilerList))
```

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down:  same here. E.g. 

```
array2DataLogs = list(map(lambda profiler: profiler.spinningBodyOutMsg.recorder(dataRecRate),
                              array2RotProfilerList))
```

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down: More mapping over a list `map(lambda logger: scSim.AddModelToTask(fswTaskName, logger), array1DataLogs)`

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down: We'll want to map over these lists again. And this will of course require the plots below to change.

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py]: :small_red_triangle_down: All these plots will likely change after the above changes to lists. We can use the same technique here.

File: examples/scenarioSolarArrayDeployment.py
Comment (patkenneally) [inline examples/scenarioSolarArrayDeployment.py:542]: :small_red_triangle_down: Need a new line at the end of the file.

**Review by patkenneally — CHANGES_REQUESTED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

Solid. We should make good use of the map function with a lambda to remove all the repetition of individual variable names

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: I can't figure out what command to use for plotting with the new lists you are suggesting I create. How do you extract the elements from the list individually here?

Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: @patkenneally

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: @patkenneally I can't figure out what command to use for plotting with the new lists you are suggesting I create. How do you extract the elements from the list individually here?

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: Done

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: Done

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: Done

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: Done

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py:542]: Done

File: examples/scenarioSolarArrayDeployment.py
Comment (leahkiner) [reply examples/scenarioSolarArrayDeployment.py]: Done

**Review by patkenneally — APPROVED** [review]

**Pull request merged**
